src/idl_gen_cpp.cpp

```diff
@@ -543,10 +553,16 @@ static void GenStruct(const Parser &parser, StructDef &struct_def,
     auto &field = **it;
     code += "  " + GenTypeGet(parser, field.value.type, " ", "", " ", false);
     code += field.name + "_;\n";
-    GenPadding(field, [&code, &padding_id](int bits) {
-      code += "  int" + NumToString(bits) +
-              "_t __padding" + NumToString(padding_id++) + ";\n";
-    });
+
+    struct PaddingDefinition
+    {
+      static void Method(int bits, std::string &code_l, int &padding_id_l) {
```

Comment:
operator() cannot be static, and std::function is unable to be initialized from a local function object.